[feat] 로그인 연결 및 메인페이지 관련 컴포넌트 작성 및 조립

shmallow1 opened the pull request:
### 📕 Issue Number

#14 
#18 
#34 
#47 

### 📙 작업 내역

> 구현 내용 및 작업 했던 내역

- [x] 로그인 연결
- [x] 메인페이지 아톰 컴포넌트 작성
- [x] 메인페이지 조립
- [x] Atom 컴포넌트들만 styled-component 수정 작업 (추후 다른 컴포넌트들에게도 작업 필요)
- [x] next.js 웹팩 설정

### 📘 작업 유형

- [x] 신규 기능 추가
- [x] 버그 수정
- [x] 리펙토링

### 📋 체크리스트

- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
      <br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 메인 페이지
![](https://ifh.cc/g/C4nGy3.jpg)

- 에러 사항 
![](https://ifh.cc/g/qgN6JB.jpg)

- 웹팩 설정을 적용시켰는데 .gql파일을 을 찾을 수 없다는 에러가 발생합니다. 그런데 값은 제대로 읽어와지는 기이한 현상이 발생하고 있습니다.. 😢 

<br/><br/>

shmallow1 committed:
[chore] graphql-tag, @ezit/next-source-maps, types/styled-components 모듈 설치

1. .gql, .graphql 파일을 직접 불러오기 위해 graphql-tag, @ezit/next-sourece-maps 모듈 설치
2. typescript에서 styled-compone을 사용하기 위해@types/styled-components 모듈 설치

shmallow1 committed:
[feat] 메인페이지에서 사용할 아이콘 아톰 컴포넌트 작성

shmallow1 committed:
[feat] 아이콘 아톰 컴포넌트에 width, height 속성 추가

아이콘의 크기를 조절하기 위해 width, height 속성을 추가

shmallow1 committed:
[feat] 메인페이지에 필요한 TextArea 아톰 컴포넌트 작성

shmallow1 committed:
[feat] 메인 페이지에 필요한 메인컨테이너 organism 컴포넌트 작성

shmallow1 committed:
[feat] 메인페이지에 필요한 NewTweetContainer Organisms 컴포넌트 작성

shmallow1 committed:
[feat] 메인 페이지에 필요한 TweetContainer Organisms 컴포넌트 작성

shmallow1 committed:
[refactor] styled components를 적용하기 위해 login.tsx 파일 삭제

shmallow1 committed:
[refactor] pages 폴더에 login폴더 생성 및 로그인 페이지 수정

1. styled.ts 파일에 styled component를 이용해 컴포넌트 생성
2. styled.ts 파일에 있는 컴포넌트를 사용해 로그인 페이지 수정

shmallow1 committed:
[feat] 메인페이지 조립

shmallow1 committed:
[refactor] Form 아톰 컴포넌트에 styled component 방식 적용

shmallow1 committed:
[refactor] ProfileImg 아톰 컴포넌트에 styled components 방식 적용

shmallow1 committed:
[refactor] SearchBar Molecule 컴포넌트에 styled-component 방식 적용

shmallow1 committed:
[refactor] UserInfo Molecule 컴포넌트 수정

1. styled-component 작성
2. onClick, width 속성 추가

shmallow1 committed:
[refactor] LoginLeftSection Organism 컴포넌트에 styled-component 적용

shmallow1 committed:
[refactor] LoginRightSection Organism 컴포넌트에 styled-component 적용

shmallow1 committed:
[feat] Input 아톰 컴포넌트에 Onchange, value 속성 추가

shmallow1 committed:
[fefactor] Icons import 구문 수정

1. Icons.index 파일을 import하여 한번에 불러오도록 수정

shmallow1 committed:
[refactor] SideBar Organisms 컴포넌트 수정

1. styled component 적용
2. Icons import 구문 수정
3. Icon들에 width, height 적용

shmallow1 committed:
[feat] onClick 속성 추가

shmallow1 committed:
[fix] Button Molecule 컴포넌트에 width 속성이 빠져있어서 추가

shmallow1 committed:
[feat] next.js에서 styled-component가 정상적으로 작동하도록 코드 추가

shmallow1 committed:
[feat] .gql, .graphql 파일을 사용할 수 있도록 설정

1. next.config.js에서 웹팩 설정
2. graphql/github.gql 파일에 쿼리 작성
3. callback/index.tsx 파일에서 gql 파일 import

joojaewoo committed:
[feat] jwt토큰 담아 요청 보내기

- jwt토큰을 담아 요청 보내기
- link와 header을 concat로 연결하여 link 설정

joojaewoo committed:
[fix] 타입 추가로 인해 .gql import 에러 수정

- @typedp .gql 파일을 선언해줌으로 callback에서 import 에러 수정

joojaewoo committed:
[refactor] _document 파일명 수정

- js 파일을 tsx 파일로 수정

joojaewoo commented:
### 추가 구현부분
- _app.tsx에서 localStorage에 있는 jwt_token 을 헤더에 담기
- .gql폴더 import시 에러 해결
  - type를 declear 해줌으로써 해결
- SearchBar에 onChange 함수 추가
- 로그인성공시 home page로 이동

joojaewoo committed:
[feat] SearchBar onChage 함수 추가

- 입력시 rendering를 위해 onChange함수 추가
- home에서 useState사용 (나중에 공통으로 사용되는 곳에 구현 하면 될듯)

joojaewoo committed:
[fix] github 버튼 URL 수정

- 깃허브 로그인 url client id 수정
- 로그인 성공시 home page로 이동

joojaewoo committed:
[fix] 에러 수정

- home에서 반복문에 key값이 없어서 뜨는 오류 수정
- key값에 index를 사용하여 eslint disabled
- github 로그인 url 오타 수정
